### ISSUE TITLE
Introduce more failpoints into etcd watch

### ISSUE THREAD

**Issue by serathius** [etcd-io/etcd#18681]
### What would you like to be added?

Spin-off from https://github.com/etcd-io/etcd/issues/17529

Similar to https://github.com/etcd-io/etcd/pull/17680/files we should add more failpoints into watch and simulate:
* slow watch sync
* slow watch resynchronization
* slow watch progress notify

etc

### Why is this needed?

Prevent and detect other issues like https://github.com/etcd-io/etcd/issues/17529

**Comment by Bharadwajshivam28** [comment]
Hey @serathius can I work on this?

**Comment by serathius** [comment]
Suer, however this task is not recommended for new contributors.

**Comment by serathius** [comment]
ping @Bharadwajshivam28

**Comment by serathius** [comment]
We just discovered issue with watch on future revision https://github.com/etcd-io/etcd/issues/20221

**Comment by akstron** [comment]
Hi @serathius / @henrybear327 , I would like to pick this up. Quick question, we just want to introduce more failpoints for `Exploratory` scenarios right?

**Comment by serathius** [comment]
Yes, for this issue we want to identify interesting places in etcd server watch code and add fail point with sleep infection for exploratory scenarios. There is separate work thread related to regression testing on reproducing historical watch issues that is being worked on by @henrybear327. Both might lead to introduction of new fail points.

**Comment by Ackberry** [comment]
Hey @serathius , seems there is no communication from the person involved since a while. If this issue still exists, I would love to take this and do it as Codepath's open source capstone project. Happy to help :)

**Comment by serathius** [comment]
Don't know "Codepath's open source capstone project" but the issue is a 3x 1 line comment change.

**Comment by Ackberry** [comment]
that shouldn't be a problem. I am supposed to do a few issues altogether for the capstone, so this would be one of them!
Let me know if that still works for you.

**Comment by darshil929** [comment]
Hi @serathius, I have raised #22044 adding the three watch failpoints discussed above, could you take a look when you get a chance?
